caput/pipeline/extensions/_configtypes.py

```diff
@@ -1,38 +1,40 @@
 """Property config types."""
 
+from __future__ import annotations
+
 from ...config import CaputConfigError, Property
 from ...memdata import fileformats
 
 __all__ = ["file_format"]
 
 
-def file_format(default: str | fileformats.FileFormat | None = None) -> Property:
+def file_format(default=None):
     """Property type that accepts only "zarr", or "hdf5".
 
-    Returns the selected `caput.fileformat.FileFormat` subclass or `caput.fileformats.HDF5` if `value == default`.
+    Returns the selected :py:class:`~caput.fileformats.FileFormat` subclass or
+    :py:class:`~caput.fileformats.HDF5` if `value` == "default".
 
     Parameters
     ----------
-    default
+    default : str | FileFormat | None, optional
         A string or type object specifying the fileformat
 
     Returns
     -------
-    prop
+    file_format : FileFormat
```

Comment:
```suggestion
    prop : Property
```
This _function_ returns a `Property`.  The `Property` is the thing that'll return a `FileFormat`.